src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -467,6 +474,65 @@ bool CascadiaSettings::_isPowerShellCoreInstalledInPath(const std::wstring_view
     return false;
 }
 
+// Function Description:
+// - Adds all of the WSL profiles to the provided container.
+// Arguments:
+// - A ref to the profiles container where the WSL profiles are to be added
+// Return Value:
+// - <none>
+void CascadiaSettings::_AppendWslProfiles(std::vector<TerminalApp::Profile>& profileStorage)
+{
+    wil::unique_handle readPipe;
+    wil::unique_handle writePipe;
+    SECURITY_ATTRIBUTES sa{ sizeof(sa), nullptr, true };
+    THROW_IF_WIN32_BOOL_FALSE(CreatePipe(&readPipe, &writePipe, &sa, 0));
+    STARTUPINFO si{};
+    si.cb = sizeof(si);
+    si.dwFlags = STARTF_USESTDHANDLES;
+    si.hStdOutput = writePipe.get();
+    si.hStdError = writePipe.get();
+    wil::unique_process_information pi{};
+    std::wstring command = L"wsl.exe --list";
+
+    THROW_IF_WIN32_BOOL_FALSE(CreateProcessW(nullptr, const_cast<LPWSTR>(command.c_str()), nullptr, nullptr,
+                                                TRUE, CREATE_NO_WINDOW, nullptr, nullptr, &si, &pi));
+    switch (WaitForSingleObject(pi.hProcess, INFINITE)) {
+    case WAIT_OBJECT_0:
+        break;
+    case WAIT_ABANDONED:
+    case WAIT_TIMEOUT:
+        THROW_HR(ERROR_CHILD_NOT_COMPLETE);
+    case WAIT_FAILED:
+        THROW_LAST_ERROR();
+    default:
+        THROW_HR(ERROR_UNHANDLED_EXCEPTION);
+    }
+    DWORD exitCode;
+    if ((GetExitCodeProcess(pi.hProcess, &exitCode) == false) || (exitCode != 0)) {
+        THROW_HR(E_INVALIDARG);
+    }
+    DWORD bytesAvailable;
+    THROW_IF_WIN32_BOOL_FALSE(PeekNamedPipe(readPipe.get(), nullptr, NULL, nullptr, &bytesAvailable, nullptr));
+    FILE* hPipe = _wfdopen(_open_osfhandle((intptr_t)readPipe.get(), _O_WTEXT | _O_RDONLY), L"r");
+        //don't call fclose on hPipe because the readPipe handle is managed by wil and this will cause an error.
+    std::wfstream pipe{ hPipe };
+    std::wstring wline;
+    std::getline(pipe, wline); //remove the header from the output.
+    while (pipe.tellp() < bytesAvailable) {
+        std::getline(pipe, wline);
+        std::wstringstream wlinestream(wline);
+        if (wlinestream) {
+            std::wstring distName;
+            std::getline(wlinestream, distName, L' ');
+            auto WSLDistro{ _CreateDefaultProfile(distName) };
+            WSLDistro.SetCommandline(L"wsl.exe -d " + distName);
+            WSLDistro.SetStartingDirectory(L"~");
+            WSLDistro.SetColorScheme({ L"Campbell" });

```

Comment:
Do we want to set the icon for these distros to the [tux penguin](https://github.com/microsoft/terminal/blob/master/src/cascadia/CascadiaPackage/ProfileIcons/%7B9acb9455-ca41-5af7-950f-6bca1bc9722f%7D.scale-200.png) that we ship with the package?

@dhowett-msft can correct me if I'm wrong, but I think that path is:
`"ms-appx:///ProfileIcons/{9acb9455-ca41-5af7-950f-6bca1bc9722f}.png"`

Reply:
I was hoping that we could determine the icon for each WSL profile, but I don't think we'll be able to using `WSL --list`.  I threw in some code to give us the penguin as a default until we figure out something better.

Reply:
I'm actually pretty sure that's not feasible at all, since some distros might be installed from the store, while others could be manually installed (and won't have an icon). So this is certainly a close enough compromise